pkg/placement/membership.go

```diff
@@ -0,0 +1,118 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package placement
+
+import (
+	"time"
+
+	"github.com/dapr/dapr/pkg/placement/raft"
+	v1pb "github.com/dapr/dapr/pkg/proto/placement/v1"
+)
+
+// MembershipChangeWorker is the worker to change the state of membership
+// and update the consistent hashing tables for actors.
+func (p *Service) MembershipChangeWorker() {
+	faultHostDetectTimer := time.NewTicker(faultyHostDetectInterval)
+	flushTimer := time.NewTicker(flushTimerInterval)
+	lastFlushTimestamp := time.Now().UTC()
+
+	hostUpdateCount := 0
+
+	for {
+		if !p.raftNode.IsLeader() {
+			// TODO: use leader election channel instead of sleep
+			// <- p.raftNode.leaderCh
+			time.Sleep(100 * time.Millisecond)
+			continue
+		}
+
+		select {
+		case op := <-p.membershipCh:
+			switch op.cmdType {
+			case raft.MemberUpsert:
+				fallthrough
+
+			case raft.MemberRemove:
+				updated, raftErr := p.raftNode.ApplyCommand(op.cmdType, op.host)
+				if raftErr != nil {
+					log.Debugf("fail to apply command: %v", raftErr)
```

Comment:
Shouldn't the worker restart with a clean state in case of error?

Reply:
+1 worker should not restart. instead, we may need to retry apply command.

Reply:
I missed why I didn't add retry here :) each runtime will send the heartbeat every 1 second. For MemberUpsert, as long as runtime sends heartbeat, it will try to add or update updateAt timestamp in Raft state every 1 second. 

For MemberRemove, the faulty host detect timer at L65 will try to remove node if MemberRemove was not successful.

so the state will be eventually consistent. but I agree to add simple linear retry here.

Reply:
added it.